File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-10.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int result = 1;
        for (int l = 0; l < s.length(); ++l) {
            if (l > 0 && s[l] == s[l - 1]) {
                continue;
            }
            int left = l;
            int right = left;
            int replacements_left = k;
            char curr_char = s[l];
            while (right + 1 < s.length() && (s[right + 1] == curr_char || replacements_left > 0)) {
                if (s[right + 1] != curr_char) {
                    --replacements_left;
                }
                ++right;
            }
            left = max(0, l - replacements_left);
            result = max(result, right - left + 1);
        }
        return result;
    }
};

File: Data Structures & Algorithms/binary-tree-diameter/submission-10.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        maxDepthLeft = (1 if root.left else 0) + self.maxDepth(root.left)
        maxDepthRight = (1 if root.right else 0) + self.maxDepth(root.right)
        diameterThroughRoot = maxDepthLeft + maxDepthRight
        self.diameter = max(self.diameter, diameterThroughRoot)
        return max(maxDepthLeft, maxDepthRight)
    def diameterOfBinaryTree(self, root: Optional[TreeNode]) -> int:
        self.diameter = 0
        self.maxDepth(root)
        return self.diameter

        